src/samplers/support/transform.jl

```diff
@@ -167,6 +198,23 @@ Distributions.logpdf{T}(d::SimplexDistribution, x::Vector{T}, transform::Bool) =
   lp
 end
 
+Distributions.logpdf{T}(d::SimplexDistribution, X::Matrix{T}, transform::Bool) = begin
+  lp = logpdf(d, X)
+  if transform
+    nrow, ncol = size(X)
+    K = nrow
+    Z = Matrix{T}(nrow - 1, ncol)
+    for k = 1:K-1
+      Z[k,:] = X[k,:] ./ (one(T) - sum(X[1:k-1,:],1))'
+    end
+    for k = 1:K-1
+      lp += log(Z[k,:]) + log(one(T) - Z[k,:]) + log(one(T) - sum(X[1:k-1,:], 1))'
+    end
+  end
+  lp
+end
+
+# REVIEW: why do we put this piece of code here?
```

Comment:
the distributions package does not check `d.p[x] > 0.0`, so I override their definition here. It's probably better to submit a PR to Distributions package to fix this.